prow/test/integration/test/webhook_server_test.go

```diff
@@ -55,11 +67,64 @@ func TestWebhookServerValidateCluster(t *testing.T) {
 	//prowjob should be created successfully
 	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absValidProwJobPath).Run()
 	if err != nil {
-		t.Errorf("prowjob was not properly validated")
+		t.Errorf("prowjob was not properly validated %v", err)
 	}
 	//cleanup prowjob that was created
 	err = cleanUpCluster(validProwJobID)
 	if err != nil {
 		t.Errorf("could not delete prowjob %s", validProwJobID)
 	}
 }
+
+func TestWebhookServerMutateProwjob(t *testing.T) {
+	validBareProwJobPath := "./testdata/valid_bare_prowjob.yaml"
+	absValidBareProwJobPath, err := filepath.Abs(validBareProwJobPath)
+	var mutatedProwJob v1.ProwJob
+	if err != nil {
+		t.Fatalf("could not find absolute file path")
+	}
+	//prowjob should be created successfully
+	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absValidBareProwJobPath).Run()
+	if err != nil {
+		t.Errorf("prowjob was not properly validated %v", err)
```

Comment:
T.Fatal is better here, `Errorf` means this test will keep going which will produce more logs that are all side effects from the failure here, those logs might cover the real root cause of failure